### DISCUSSION TITLE
Wrong trending arrows

### DISCUSSION THREAD

**Discussion by JohhnyP** [NightscoutFoundation/xDrip#3198] [Bug reports]
Hello.

I have the following setup.
- **Descom G1**
- **NS 15.0.2**
- **xDrip** (Collects data, uplaod to NS, broadcasts locally)
![image](https://github.com/NightscoutFoundation/xDrip/assets/11662759/2050fd08-d414-469a-861b-76607b4c94c7)

- **AAPS** (Reads local broadcasts from xDrip)
- **AAPS** Clients (Reads from NS)

The trending arrows are not corresponding to the correct values
THe isue started yesterday when i switched from L2 to Descom and applied the relevant config changes on xDrip.

![image](https://github.com/NightscoutFoundation/xDrip/assets/11662759/87ab6a40-93c3-4627-8086-e28efe65f8fb)

**Comment by Navid200** [comment]
When you use xDrip as a collector from Dexcom One, I believe the trend arrow xDrip shows comes from the Dexcom One transmitter.  
I don't believe xDrip is calculating the trend arrow in that case.

**Comment by JohhnyP** [comment]
Thanks for the response @Navid200 

1) The arrows in master xDrip are correct.
![xd](https://github.com/NightscoutFoundation/xDrip/assets/11662759/c53bfb86-06da-40c8-b2e1-3e283f2f463b)

2) They are also correct in master AAPS (since it's getting the locally broadcasted values from xDrip)
![xd3](https://github.com/NightscoutFoundation/xDrip/assets/11662759/6a9ab70a-e3ec-42cd-9d69-1c251b985901)

3) However the xDrip BG Data table is wrong (most likely becasue what you've said, they're the trending arrows from G1)
![xd2](https://github.com/NightscoutFoundation/xDrip/assets/11662759/454132b5-c072-4d62-ab7b-89d5e3f9df5a)

4) But it seems these (the wrong ones) are the trending arrows uploaded to NS
![xd4](https://github.com/NightscoutFoundation/xDrip/assets/11662759/7bbcb915-b1c8-4f90-b3f5-11d0f031bc6c)

5) As a result anyone who fetches data from NS (APPS Client, NSFollowers etc have the wrong arrows)
![xd5](https://github.com/NightscoutFoundation/xDrip/assets/11662759/c7485162-6329-447d-9712-401ec5443630)

If i understand correctly, It seems that xDrip uploads the trending arrow received from G1 (wrong) and the one calculated (correct).

**Comment by jonprogers** [comment]
I use G6 rather than Dex1 although I see differing trend arrows too. I believe that this is down to different calculation algorithms so rather than one being 'right' and the other being 'wrong' they are both correct based on the algorithm they are calculated by.

The black box transmitter algorithm from Dexcom appears to calculate a trend over a longer period of time than xDrip which results in the Dex trend tending to be smoother and the xDrip trend tending to be more responsive to change. They are reporting against a different definition of 'trend'. I haven't noticed a trend that appears to be in error when looked at with this context.

I find both trends useful at different times so appreciate the fact that they both exist. There have been other discussions about this in the past so you may find better descriptions there.

**Reply by JohhnyP** [reply]
How do you find it useful exactly?

Previous value was 160, there was a -9 decrease and now xDrip show diagonal arrow down -9 and NS Diagonal Arrow up

**Reply by jonprogers** [reply]
Because I get a short term trend from xDrip and a longer term trend from Dexcom, if they correlate then I have more confidence in the stability of the sensor and if they differ then I see that as an indicator that my readings are currently a bit noisy or that something is changing. 

In your example to get the the arrows you describe then the 160 -> 151 could be part of a sequence 130 -> 140 -> 150 -> 160 -> 151. In me (we are all different!) there is a good chance my next reading could be 180 or equally it could be 145 and the fact that the trends are different can alert me to this uncertainty.
